autoload/neomake.vim

```diff
@@ -102,12 +106,14 @@ function! s:MakeJob(make_id, maker) abort
         call add(args, '%:p')
     endif
 
+    call neomake#utils#DebugMessage('Arguments before expansion: '. string(args))
     if neomake#utils#IsRunningWindows()
-        " Don't expand &shellcmdflag argument of cmd.exe
-        call map(args, 'v:val !=? &shellcmdflag ? expand(v:val) : v:val')
+        " Don't expand &shellcmdflag argument of cmd.exe nor arguments like '"something"'
+        call map(args, "(v:val ==? &shellcmdflag || v:val =~ '.*\".*\".*') ? v:val : expand(v:val)")
     else
```

Comment:
Would be nice to have tests for this, even when they only run on Windows then.
Feel free to factor it out into its own function for that.